airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,475 @@
+                                    'name' : self.mesos_docker_volume_dag_name
+                                }
+                            }
+                        }
+                    ],
+                    'docker': {
+                        'image': self.mesos_slave_docker_image,
+                        'force_pull_image': 'true',
+                        'privileged' : 'true',
+                        'parameters' : [
+                            {
+                                'key' : 'volume',
+                                'value' : self.mesos_docker_sock + ':/var/run/docker.sock'
+                            }
+                        ]
+                    }
+                }
+            }
+
+            option = {
+                'Filters' : {
+                    'RefuseSeconds': '0.5'
+                }
+            }
+
+            tasks.append(task)
+            remaining_cpus -= self.task_cpu
+            remaining_mem -= self.task_mem
+        mesos_offer.accept(tasks, option)
+
+    def subscribed(self, driver):
+        """
+        Subscribe to Mesos Master
+
+        :param driver: Mesos driver object
+        """
+        from airflow.models import Connection
+
+        # Update the Framework ID in the database.
+        session = Session()
+        conn_id = FRAMEWORK_CONNID_PREFIX + get_framework_name()
+        connection = Session.query(Connection).filter_by(conn_id=conn_id).first()
+        if connection is None:
+            connection = Connection(
+                conn_id=conn_id,
+                conn_type='mesos_framework-id',
+                extra=driver.frameworkId
+            )
+        else:
+            connection.extra = driver.frameworkId
+
+        session.commit()
+        self.driver = driver
+        Session.remove()
+
+    def status_update(self, update):
+        """Update the Status of the Tasks. Based by Mesos Events."""
+        self.log.info(
+            "Task %s is in state %s", update['status']['task_id']['value'], update['status']['state'])
+
+        try:
+            key = self.task_key_map[update['status']['task_id']['value']]
+
+        except KeyError:
+            # The map may not contain an item if the framework re-registered
+            # after a failover.
+            # Discard these tasks.
+            self.log.info("Unrecognised task key %s", update['status']['task_id']['value'])
+            return
+
+        if update['status']['state'] == "TASK_RUNNING":
+            self.result_queue.put((key, State.RUNNING))
+            return
+
+        if update['status']['state'] == "TASK_FINISHED":
+            self.result_queue.put((key, State.SUCCESS))
+            return
+
+        if update['status']['state'] == "TASK_LOST" or \
+           update['status']['state'] == "TASK_KILLED" or \
+           update['status']['state'] == "TASK_FAILED":
+            self.result_queue.put((key, State.FAILED))
+            return
```

Comment:
How about adding at the beginning of this function:
```
status = update["status"]
```
and then creating following variables
```
value = status["task_id"]["value"]
state = status["state"]
```
I think this my simplify the code

Reply:
Also, how about creating enum class for task states?

Reply:
ok, now I got it. :-) yeah I think u are right it would be easier to read the code.

Reply:
But I will change it like that. What do u think?
```
task_id = update["status"]["task_id"]["value"]
task_state = update["status"]["state"]

```

Reply:
Even better! 👌